applications/HDF5Application/custom_io/hdf5_file.cpp

```diff
@@ -336,32 +381,42 @@ std::vector<std::string> File::GetLinkNames(const std::string& rGroupPath) const
     }
     KRATOS_ERROR_IF(H5Gclose(group_id) < 0) << "H5Gclose failed." << std::endl;
     return names;
+
     KRATOS_CATCH("");
 }
 
 std::vector<std::string> File::GetGroupNames(const std::string& rGroupPath) const
 {
     KRATOS_TRY;
+
     std::vector<std::string> names;
     std::vector<std::string> link_names = GetLinkNames(rGroupPath);
     names.reserve(link_names.size());
-    for (const auto& r_name : link_names)
-        if (IsGroup(rGroupPath + '/' + r_name))
+    for (const auto& r_name : link_names) {
+        if (IsGroup(rGroupPath + '/' + r_name)) {
             names.push_back(r_name);
```

Comment:
you won't ever reuse the items in `link_names` so you might as well `move` them and prevent a copy.

```suggestion
    for (auto& r_name : link_names) {
        if (IsGroup(rGroupPath + '/' + r_name)) {
            names.emplace_back(std::move(r_name));
```

Reply:
done :)